controllers/userController.js

```diff
@@ -137,24 +128,47 @@ module.exports = {
   },
   addAddress: async (req, res) => {
     try {
-      const { address, city, state, pinCode, phoneNum, type } = req.body;
-      const data = req.tokenData;
-
-      await Address.create({
-        user_id: data.user_id,
+      const { address, city, state, pin_code, phone_no, userId } = req.body;
+      const addAddress = await addresses.create({
         address,
         city,
         state,
-        pinCode,
-        phoneNum,
-        type,
+        pin_code,
+        phone_no,
+        userId,
       });
+      res.status(200).send(addAddress);
+    } catch (err) {
+      console.log("Error occurred:", err);
+      res.status(500).send({ message: "Internal server error" });
+    }
+  },
+  deleteAddress: async (req, res) => {
+    try {
+      const data = req.body.addressArray;
+      for (const addressId of data) {
+        const user_address = await Address.findOne({
+          where: {
+            user_id: req.params.id,
```

Comment:
why do you need user_id for delete an address, you can directly delete it using id @akaashinno